src/main/java/frc/robot/poseestimator/WPILibPoseEstimator/WPILibPoseEstimator.java

```diff
@@ -0,0 +1,105 @@
+package frc.robot.poseestimator.WPILibPoseEstimator;
+
+import edu.wpi.first.math.estimator.PoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.Odometry;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import frc.robot.poseestimator.IPoseEstimator;
+import frc.robot.poseestimator.PoseEstimationMath;
+import frc.robot.poseestimator.observations.OdometryObservation;
+import frc.robot.subsystems.GBSubsystem;
+import frc.robot.vision.data.AprilTagVisionData;
+
+import java.util.List;
+import java.util.Optional;
+
+public class WPILibPoseEstimator extends GBSubsystem implements IPoseEstimator {
+
+	private final PoseEstimator<SwerveModulePosition[]> poseEstimator;
+	private final Odometry<SwerveModulePosition[]> odometryEstimator;
+
+	public WPILibPoseEstimator(String logPath, SwerveDriveKinematics kinematics, SwerveModulePosition[] modulePositions) {
+		super(logPath);
+		this.poseEstimator = new PoseEstimator<>(
+			kinematics,
+			new Odometry<>(
+				kinematics,
+				WPILibPoseEstimatorConstants.STARTING_ODOMETRY_ANGLE,
+				modulePositions,
+				WPILibPoseEstimatorConstants.STARTING_ODOMETRY_POSE
+			),
+			WPILibPoseEstimatorConstants.DEFAULT_ODOMETRY_STANDARD_DEVIATIONS.getWPILibStandardDeviations(),
+			WPILibPoseEstimatorConstants.DEFAULT_VISION_STANDARD_DEVIATIONS.getWPILibStandardDeviations()
+		);
+		this.odometryEstimator = new Odometry<>(
+			kinematics,
+			WPILibPoseEstimatorConstants.STARTING_ODOMETRY_ANGLE,
+			modulePositions,
+			WPILibPoseEstimatorConstants.STARTING_ODOMETRY_POSE
+		);
+	}
+
+
+	@Override
+	public void resetPose(Pose2d newPose) {
+		poseEstimator.resetPose(newPose);
+	}
+
+	@Override
+	public Pose2d getEstimatedPose() {
+		return poseEstimator.getEstimatedPosition();
+	}
+
+	@Override
+	public Pose2d getEstimatedPoseAtTimestamp(double timestamp) {
+		return poseEstimator.sampleAt(timestamp).orElseGet(this::getEstimatedPose);
+	}
+
+	@Override
+	public void updateOdometry(OdometryObservation[] odometryObservations) {
+		for (OdometryObservation odometryObservation : odometryObservations) {
+			poseEstimator.update(odometryObservation.gyroAngle(), odometryObservation.wheelPositions());
+			updateOdometryPose(odometryObservation);
+		}
+	}
+
+	@Override
+	public void resetOdometry(SwerveModulePosition[] wheelPositions, Rotation2d gyroAngle, Pose2d robotPose) {
+		poseEstimator.resetPosition(gyroAngle, wheelPositions, robotPose);
+		odometryEstimator.resetPosition(gyroAngle, wheelPositions, robotPose);
+	}
+
+	@Override
+	public Pose2d getOdometryPose() {
+		return odometryEstimator.getPoseMeters();
+	}
+
+	@Override
+	public void setHeading(Rotation2d newHeading) {
+		poseEstimator.resetRotation(newHeading);
+		odometryEstimator.resetRotation(newHeading);
+	}
+
+	@Override
+	public void updateVision(List<AprilTagVisionData> robotPoseVisionData) {}
```

Comment:
Why is the pose estimator ignoring vision measurements?
Even if at the moment there is no vision I would still expect it to implement it's side of it
Especially as it is simply wrapping the wpilib addvisionmeasurement

Reply:
Okay just read the function below still dint understand why this is empty but guess there is a reason

Reply:
there isn't any reason

we just forgot to implement it. I catch it now in testing, prev testing branches have had this method implemented